packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -1,12 +1,359 @@
+            "l1Token",
+            _relayData.l1Token
+        );
+        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
+            intermediateAncillaryData,
+            "amount",
+            _relayData.amount
+        );
+        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
+            intermediateAncillaryData,
+            "maxFeePct",
+            _relayData.maxFeePct
+        );
+        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
+            intermediateAncillaryData,
+            "proposerRewardPct",
+            _relayData.proposerRewardPct
+        );
+        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
+            intermediateAncillaryData,
+            "realizedFeePct",
+            _relayData.realizedFeePct
+        );
+        intermediateAncillaryData = AncillaryData.appendKeyValueAddress(
+            intermediateAncillaryData,
+            "slowRelayer",
+            _relayData.slowRelayer
+        );
+
+        // Add global state data stored by this contract:
+        intermediateAncillaryData = AncillaryData.appendKeyValueAddress(
+            intermediateAncillaryData,
+            "depositContract",
+            bridgePoolFactory.depositContract()
+        );
+
+        return intermediateAncillaryData;
+    }
+
+    /**************************************
+     *        INTERNAL FUNCTIONS          *
+     **************************************/
+
+    function _getOptimisticOracle() private view returns (OptimisticOracleInterface) {
+        return
+            OptimisticOracleInterface(
+                FinderInterface(bridgePoolFactory.finder()).getImplementationAddress(OracleInterfaces.OptimisticOracle)
+            );
+    }
+
+    function _getStore() private view returns (StoreInterface) {
+        return
+            StoreInterface(
+                FinderInterface(bridgePoolFactory.finder()).getImplementationAddress(OracleInterfaces.Store)
+            );
+    }
+
+    function _requestOraclePriceRelay(
+        address l1Token,
+        uint256 amount,
+        uint256 requestTimestamp,
+        bytes memory customAncillaryData,
+        uint256 proposerRewardPct
+    ) private {
+        OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
+        uint256 proposerBondPct =
+            FixedPoint.Unsigned(bridgePoolFactory.proposerBondPct()).div(FixedPoint.fromUnscaledUint(1)).rawValue;
+
+        // Optimistic oracle will pull proposer reward from passive LP pool.
+        uint256 proposerReward =
+            FixedPoint
+                .Unsigned(proposerRewardPct)
+                .div(FixedPoint.fromUnscaledUint(1))
+                .mul(FixedPoint.Unsigned(amount))
+                .rawValue;
+
+        // Sanity check that pool balance is enough to cover relay amount + proposer reward.
+        require(IERC20(l1Token).balanceOf(address(this)) >= amount + proposerReward, "Insufficient pool balance");
```

Comment:
Does amount include the fees or exclude the fees?

Reply:
doesn't include fees, but fees can only decrease the amount of funds to pull from the pool so this check is conservative